gameplay/gameplay.go

```diff
@@ -0,0 +1,98 @@
+package gameplay
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"reflect"
+	"regexp"
+	"time"
+)
+
+var guesses int
+
+func GetDifficulty() int {
+	var difficulty int
+
+	for {
+		fmt.Print("Please select a difficulty: ")
+		fmt.Print("Easy: 1 ")
+		fmt.Print("Medium: 2 ")
+		fmt.Print("Hard: 3 ")
+		fmt.Scan(&difficulty)
+		if validatedDifficulty(difficulty) {
+			return setGuesses(difficulty)
+		} else {
+			fmt.Println("Sorry, it must be a number between 1 and 3")
+
+		}
+	}
+
+	return difficulty
+}
+
+func validatedDifficulty(d int) bool {
+
+	//check to see if input is type int
+	//used Sprint to cast interface to string so I can compare
+	if fmt.Sprint(reflect.TypeOf(d)) == "int" && d <= 3 {
+
+		if regexp.MustCompile(`1`).MatchString(fmt.Sprint(d)) || regexp.MustCompile(`2`).MatchString(fmt.Sprint(d)) || regexp.MustCompile(`3`).MatchString(fmt.Sprint(d)) {
```

Comment:
You should use the strings.Itoa to convert a string to an integer. 
You should also use double quotes for strings instead of backticks

Back-ticks are typically only used for multiline strings or when you need to place double quotes in a string